datacapturing/src/androidTest/java/de/cyface/datacapturing/persistence/PersistenceLayerTest.java

```diff
@@ -0,0 +1,181 @@
+     */
+    private CapturingPersistenceBehaviour capturingBehaviour;
+
+    /**
+     * Initializes the <code>oocut</code> with the Android persistence stack.
+     */
+    @Before
+    public void setUp() {
+        context = InstrumentationRegistry.getInstrumentation().getTargetContext();
+        resolver = context.getContentResolver();
+        this.capturingBehaviour = new CapturingPersistenceBehaviour();
+        oocut = new PersistenceLayer(context, resolver, AUTHORITY, capturingBehaviour);
+    }
+
+    /**
+     * Deletes all content from the content provider, to leave the next test with a clean test environment.
+     */
+    @After
+    public void tearDown() {
+        clearPersistenceLayer(context, resolver, AUTHORITY);
+        oocut.shutdown();
+    }
+
+    /**
+     * Inserts two measurements into the database; one finished and one still running and checks, that the
+     * <code>loadFinishedMeasurements</code> method returns a list of size 1.
+     *
+     * @throws NoSuchMeasurementException Fails the test if anything unexpected happens.
+     */
+    @Test
+    public void testLoadFinishedMeasurements_oneFinishedOneRunning()
+            throws NoSuchMeasurementException, CursorIsNullException {
+        oocut.newMeasurement(Vehicle.UNKNOWN);
+        assertThat(oocut.hasMeasurement(MeasurementStatus.OPEN), is(equalTo(true)));
+        capturingBehaviour.updateRecentMeasurement(FINISHED);
+        assertThat(oocut.hasMeasurement(MeasurementStatus.OPEN), is(equalTo(false)));
+        oocut.newMeasurement(Vehicle.UNKNOWN);
+        assertThat(oocut.hasMeasurement(MeasurementStatus.OPEN), is(equalTo(true)));
+        assertThat(oocut.loadMeasurements(MeasurementStatus.FINISHED).size(), is(equalTo(1)));
+    }
+
+    /**
+     * Checks that calling {@link PersistenceLayer#loadMeasurements(MeasurementStatus)} on an empty database
+     * returns an empty list.
+     */
+    @Test
+    public void testLoadFinishedMeasurements_noMeasurements() throws CursorIsNullException {
+        assertThat(oocut.loadMeasurements(MeasurementStatus.FINISHED).isEmpty(), is(equalTo(true)));
+    }
+
+    /**
+     * Test that loading an open and a closed measurement works as expected.
+     *
+     */
+    @Test
+    public void testLoadMeasurementSuccessfully() throws NoSuchMeasurementException, CursorIsNullException {
+        final Measurement measurement = oocut.newMeasurement(Vehicle.UNKNOWN);
+        Measurement loadedOpenMeasurement = oocut.loadMeasurement(measurement.getIdentifier());
+        assertThat(loadedOpenMeasurement, is(equalTo(measurement)));
+
+        capturingBehaviour.updateRecentMeasurement(FINISHED);
+        List<Measurement> finishedMeasurements = oocut.loadMeasurements(FINISHED);
+        assertThat(finishedMeasurements.size(), is(equalTo(1)));
+        assertThat(finishedMeasurements.get(0).getIdentifier(), is(equalTo(measurement.getIdentifier())));
```

Comment:
Because it will never happen in the code that a FINISHED measurement is created out of the blue. Also, an OPEN measurement is unlikely to be loaded. As a consequence I create an OPEN as it would happen in the code, then finish this measurement and then load it as FINISHED measurement as we usually do to synchronize them.

Reply:
But I added more docu on this